--- wxWidgets-2.8.12-galaxql/src/common/filefn.cpp	2011-03-02 10:39:44.000000000 -0800
+++ wxMSW-2.8.12/src/common/filefn.cpp	2014-12-14 00:40:22.000000000 -0800
@@ -1233,6 +1233,8 @@
   #else
     if ( mkdir(wxFNCONV(dirname), perm) != 0 )
   #endif
+#elif defined(__MINGW32__)
+    if ( mkdir(wxFNCONV(dirname)) != 0 )
 #elif defined(__OS2__)
     wxUnusedVar(perm);
     if (::DosCreateDir((PSZ)dirname, NULL) != 0) // enhance for EAB's??
@@ -1274,6 +1276,8 @@
 #elif defined(__WXPALMOS__)
     // TODO with VFSFileRename()
     return false;
+#elif defined(__MINGW32__)
+    return (rmdir(OS_FILENAME(dir)) == 0);
 #else
     return (wxRmDir(OS_FILENAME(dir)) == 0);
 #endif
